### PR TITLE
[HOPSWORKS-2177] Refactor storage connectors

### PR DESCRIPTION
**Pull request by SirOibaf**

### CODE REVIEW

File: java/src/main/java/com/logicalclocks/hsfs/FeatureStore.java [hunk 1]
```diff
@@ -121,9 +121,8 @@ public OnDemandFeatureGroup getOnDemandFeatureGroup(String name) throws FeatureS
     return SparkEngine.getInstance().sql(query);
   }
 
-  public StorageConnector getStorageConnector(String name, StorageConnectorType type)
-      throws FeatureStoreException, IOException {
-    return storageConnectorApi.getByNameAndType(this, name, type);
+  public StorageConnector getStorageConnector(String name) throws FeatureStoreException, IOException {
+    return storageConnectorApi.getByNameAndType(this, name);
```
Comment (moritzmeister) [inline java/src/main/java/com/logicalclocks/hsfs/FeatureStore.java:125]: Can we call it `getByName` to make it less confusing

```suggestion
    return storageConnectorApi.getByName(this, name);
```

File: java/src/main/java/com/logicalclocks/hsfs/engine/Utils.java [hunk 2]
```diff
@@ -116,7 +114,7 @@ public String getFgName(FeatureGroup featureGroup) {
 
   public String getHiveMetastoreConnector(FeatureGroup featureGroup) throws IOException, FeatureStoreException {
     StorageConnector storageConnector = storageConnectorApi.getByNameAndType(featureGroup.getFeatureStore(),
```
Comment (moritzmeister) [inline java/src/main/java/com/logicalclocks/hsfs/engine/Utils.java:116]:
```suggestion
    StorageConnector storageConnector = storageConnectorApi.getByName(featureGroup.getFeatureStore(),
```

File: java/src/main/java/com/logicalclocks/hsfs/metadata/StorageConnectorApi.java [hunk 3]
```diff
@@ -36,7 +35,7 @@
 
   private static final Logger LOGGER = LoggerFactory.getLogger(StorageConnectorApi.class);
 
-  public StorageConnector getByNameAndType(FeatureStore featureStore, String name, StorageConnectorType type)
+  public StorageConnector getByNameAndType(FeatureStore featureStore, String name)
```
Comment (moritzmeister) [inline java/src/main/java/com/logicalclocks/hsfs/metadata/StorageConnectorApi.java:38]:
```suggestion
  public StorageConnector getByName(FeatureStore featureStore, String name)
```

File: python/hsfs/storage_connector.py [hunk 4]
```diff
@@ -110,77 +110,167 @@ def connector_type(self):
     @property
     def access_key(self):
         """Access key."""
-        return self._access_key
+        if self._storage_connector_type == "S3":
```
Comment (moritzmeister) [inline python/hsfs/storage_connector.py:113]: The user is not able to set these properties in any way right? they are always coming in capital letters from the backend?
Just want to make sure we don't need to lower/upper case these things.

File: python/hsfs/storage_connector.py [hunk 5]
```diff
@@ -190,17 +280,59 @@ def iam_role(self):
     @property
     def expiration(self):
         """Cluster temporary credential expiration time."""
-        return self._expiration
+        if (
+            self._storage_connector_type == "S3"
+            or self._storage_connector_type == "REDSHIFT"
+        ):
```
Comment (moritzmeister) [inline python/hsfs/storage_connector.py:286]: simplify

```suggestion
        if self._storage_connector_type in ["S3", "REDSHIFT"]:
```

**Review by moritzmeister — CHANGES_REQUESTED** [review]
Some naming changes

File: python/hsfs/storage_connector.py (hunk 4 above)
Comment (SirOibaf) [reply python/hsfs/storage_connector.py:113]: Yes, but it's easy enough to add an upper case. Let me do it .

Comment (SirOibaf) [reply python/hsfs/storage_connector.py]: Done

**Review by moritzmeister — APPROVED** [review]
:+1:  just make sure you have replaced the `getByNameAndType` everywhere with `getByName`, I only marked the occurences that I saw here in the code that the PR changes.

**Pull request merged**